src/lib/game.ts

```diff
@@ -87,14 +87,16 @@ export function generateEmojiArt(
   const medals = guessModeMedal(guessMode);
   const count = gameLost(solution, rows) ? "X" : rows.length;
   let result = `${name}${medals} ${gameDay + 1} ${count}/${ROW_COUNT}\n`;
-  for (let i = 0; i < rows.length; ++i) {
-    const row = rows[i];
-    for (const x of findRowStates(solution, row)) {
+  for (const [i, row] of rows.entries()) {
+    const rowStates = findRowStates(solution, row);
+    for (const x of rowStates) {
       result += x === State.Correct ? "🟩" : x === State.Present ? "🟨" : "⬛";
     }
-    if (i < rows.length - 1) {
+    
+    // Only include Discord spoilers and new lines if we haven't won yet
+    if (!rowStates.every((s) => s === State.Correct)) {
```

Comment:
This could be written as just:

```suggestion
    if (row !== solution) {
```

Also means you don't have to move the `findRowStates` away from the loop

Reply:
+1